src/main/scala/it/unibo/authsim/library/dsl/HashFunction.scala

```diff
@@ -1,24 +1,18 @@
 package it.unibo.authsim.library.dsl
 
-import com.google.common.hash.Hashing
-import com.google.common.io.BaseEncoding
-import it.HashFunction
-
 import java.nio.charset.StandardCharsets
-import java.security.MessageDigest
-import scala.util.Try
+import org.apache.commons.codec.binary.Hex
+import org.apache.commons.codec.digest.DigestUtils
 
 trait HashFunction {
-  //Marica non ha messo nulla in input ma come faccio ad hashare un valore che non ho?
   def hash(str: String): String
 }
+
 object HashFunction {
   case class SHA() extends HashFunction {
```

Comment:
What kind of algorithm SHA is it? SHA2 ( sha256, sha512, …) or SHA1 ?

Reply:
I am refactoring it right now